build/source/dshare/popMetadat.f90

```diff
@@ -751,188 +773,214 @@ subroutine read_output_file(err,message)
   if(err/=0)then; message=trim(message)//trim(cmessage); return; end if
   nWords = size(lineWords)
 
-  ! user cannot control time output
-  if (trim(lineWords(1))=='time') cycle
+  ! define variable name
+  varName = trim(lineWords(nameIndex))
 
-  ! read output frequency
-  read(lineWords(freqIndex),*,iostat=err) oFreq
-  if(err/=0)then
-   message=trim(message)//'problem reading the output frequency: check format of model control file "'//trim(outfile)//'"'
-   err=20; return
-  endif
+  ! user cannot control time output
+  if (trim(varName)=='time') cycle
 
   ! --- variables with multiple statistics options --------------------------
 
   ! idenify the data structure for the given variable (structName) and the variable index (vDex)
-  call get_ixUnknown(trim(lineWords(nameIndex)),structName,vDex,err,cmessage)
-  if (err/=0) then; message=trim(message)//trim(cmessage)//trim(linewords(nameIndex)); return; end if;
+  call get_ixUnknown(trim(varName),structName,vDex,err,cmessage)
+  if (err/=0) then; message=trim(message)//trim(cmessage)//trim(varName); return; end if;
+
+  ! --- identify the desired frequency in the metadata structure  -----------
+
+  ! process time-varying variables
+  select case(trim(structName))
+   case('indx','forc','prog','diag','flux','bvar','deriv')
+
+    ! * ensure that the frequency index exists for time varying variables
+    if(nWords<freqIndex)then
+     message=trim(message)//'must define desired output frequency for time-varing output: variable='//trim(varName)
+     err=20; return
+    endif
+
+    ! * define the frequency name
+    freqName = trim(lineWords(freqIndex))
+
+    ! * identify index in  vector
+    select case(freqName)
+     ! define cases where temporal aggregation is numeric
+     case('1');    iFreq = iLookFreq%timestep  ! assume timestep-level output
+     case('24');   iFreq = iLookFreq%day       ! assume daily output
+     ! define cases where temporal aggregation is defined using a text string
+     case default; iFreq = get_ixfreq(freqName)
+    end select
+
+    ! * check that we could find the index
+    if(iFreq<0 .or. iFreq>maxvarFreq)then
+     message=trim(message)//'unable to identify desired output frequency for variable '//trim(varName)&
+                          //' [entered "'//trim(freqName)//'"]'
+     err=20; return
+    endif
+
+   ! temporally constant variables use timestep-level output (no aggregation)
+   case default
+    iFreq    = iLookFREQ%timestep
+    freqName = '[notNeeded; use timestep]'
```

Comment:
Odd name for a frequency. Why not assign `timestep`. May also be better to define this string in a module where all the other `freqName` options are defined rather than burying it deep in the code.

Reply:
Fixed